### PR TITLE
fix getYForTopText

### PR DESCRIPTION
**Pull request by rvilarl**
Migration visual issues resolved. @0xfe you were right a proble with `getYForTopText` but in `StemmableNote`

### CODE REVIEW

File: src/stemmablenote.ts [hunk 1]
```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
 
   /** Gets the `y` value for the top modifiers at a specific `textLine`. */
   getYForTopText(textLine: number): number {
-    if (!this.stave) throw new Vex.RERR('NoStave', 'No stave attached to this note.');
```
Comment (0xfe) [inline src/stemmablenote.ts:197]: Let's keep this so we don't need the `?` below.

File: src/stemmablenote.ts [hunk 2]
```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
...
     const extents = this.getStemExtents();
-    if (extents) {
+    if (this.hasStem()) {
       return Math.min(
-        this.stave.getYForTopText(textLine),
-        extents.topY - this.render_options.annotation_spacing * (textLine + 1)
+        this.stave?.getYForTopText(textLine) ?? 0,
```
Comment (0xfe) [inline src/stemmablenote.ts:200]: If you keep the `if (!this.stave)` above, you don't need the `stave?` or the `?? 0` -- these are silent failures.

File: src/stemmablenote.ts [hunk 3]
```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
...
+        extents?.topY ?? 0 - this.render_options.annotation_spacing * (textLine + 1)
       );
     } else {
-      return this.stave.getYForTopText(textLine);
+      return this.stave?.getYForTopText(textLine) ?? 0;
```
Comment (0xfe) [inline src/stemmablenote.ts:204]: Same here.

File: src/stemmablenote.ts [hunk 4]
```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
 
   /** Gets the `y` value for the top modifiers at a specific `textLine`. */
   getYForTopText(textLine: number): number {
-    if (!this.stave) throw new Vex.RERR('NoStave', 'No stave attached to this note.');
     const extents = this.getStemExtents();
-    if (extents) {
+    if (this.hasStem()) {
```
Comment (0xfe) [inline src/stemmablenote.ts:198]: `if (!extents) throw new Vex.RERR('InvalidState', 'Stem does not have extents.');`

then we can avoid `extents?` and `??` below.

File: src/stemmablenote.ts (hunk 1 above)
Comment (rvilarl) [reply src/stemmablenote.ts:199]: ok

File: src/stemmablenote.ts (hunk 3 above)
Comment (rvilarl) [reply src/stemmablenote.ts]: done

File: src/stemmablenote.ts (hunk 4 above)
Comment (rvilarl) [reply src/stemmablenote.ts:200]: Ok I have also moved `getStemExtents` inside the `if`.

File: src/stemmablenote.ts (hunk 2 above)
Comment (rvilarl) [reply src/stemmablenote.ts]: done

**Comment by rvilarl** [comment]
Done, ready to checkin!

**Comment by 0xfe** [comment]
Great!

**Pull request merged**